courses/models.py

```diff
@@ -0,0 +1,51 @@
+from django.db import models
+
+# Create your models here.
+
+class Languages(models.Model):
+    language_name = models.CharField(max_length=20)
+
+    def __str__(self):
+        return self.language_name
+
+
+class Domains(models.Model):
+    domain_name = models.CharField(max_length=20)
+
+    def __str__(self):
+        return self.domain_name
+
+class KnowledgeBase(models.Model):
+    SKILL_BEGINNER = 'beg'
+    SKILL_INTERMEDIATE = 'int'
+    SKILL_ADVANCED = 'adv'
+
+    skill_levels = (
+        (SKILL_BEGINNER, 'BEGINNER'),
+        (SKILL_INTERMEDIATE, 'INTERMEDIATE'),
+        (SKILL_ADVANCED, 'ADVANCED')
+    )
+
+    title = models.CharField(max_length=50, null=False, blank=False)
+    description = models.TextField(null=False, blank=True)    #if no description, store empty string
+    language = models.ManyToManyField(Languages)
```

Comment:
use related names. check out the django documentation. Also the field name can be plural since many2many for readability sakes